private/buf/cmd/buf/command/beta/studioagent/studioagent.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package studioagent
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufstudioagent"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/interrupt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+)
+
+const (
+	portFlagName              = "port"
+	disallowedHeadersFlagName = "disallowed-header"
+	forwardHeadersFlagName    = "forward-header"
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <origin>",
+		Short: "Run an HTTP server as the studio agent with the origin be set as the allowed origin for CORS options.",
+		Args:  cobra.ExactArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+			bufcli.NewErrorInterceptor(),
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Port              string
+	DisallowedHeaders []string
+	ForwardHeaders    []string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Port,
+		portFlagName,
+		"8080",
+		"The port to be exposed to accept HTTP request.",
+	)
+	flagSet.StringSliceVar(
+		&f.DisallowedHeaders,
+		disallowedHeadersFlagName,
+		nil,
+		`The headers to be disallowed via the agent to the target server. Multiple headers are appended if specified multiple times.`,
+	)
+	flagSet.StringSliceVar(
+		&f.ForwardHeaders,
+		forwardHeadersFlagName,
+		nil,
+		`The headers to be forwarded via the agent to the target server. Must be a colon-separated key-value pair (like --forward-header=fromHeader1:toHeader1). Multiple header pairs are appended if specified multiple times.`,
```

Comment:
key/values are typically separated by `=` instead of `:`, and there's also a function to use to do this https://pkg.go.dev/github.com/spf13/pflag#StringToStringVar